### PR TITLE
Drag and Drop in Profile Picture

### PR DESCRIPTION
**Pull request by waseem-chamaa**

### CODE REVIEW

File: src/services [hunk 1]
```diff
@@ -1,55 +1,55 @@
-p({"class":"foam.nanos.boot.NSpec", "name":"nSpecDAO",                    "lazy":true,  "serve":true,  "serviceClass":"foam.dao.ProxyDAO", "client":"{\"of\":\"foam.nanos.boot.NSpec\"}"})
```
Comment (adamvy) [inline src/services]: Please make this change in a separate PR as it looks unrelated.

File: src/foam/nanos/auth/ProfilePictureView.js [hunk 2]
```diff
@@ -132,14 +192,62 @@ foam.CLASS({
...
+      this.isDrag = false;
+      this.data = null;
+    },
+    function onDragOver(e) {
+      this.isDrag = true;
+      e.preventDefault();    
+    },
+    function onDrop(e) {
+      e.preventDefault();  
+      this.isDrag = false;
```
Comment (adamvy) [inline src/foam/nanos/auth/ProfilePictureView.js]: I think a name like dragActive would be better than isDrag.

File: src/foam/nanos/auth/ProfilePictureView.js [hunk 3]
```diff
@@ -132,14 +192,62 @@ foam.CLASS({
...
+      if(this.uploadHidden){
```
Comment (adamvy) [inline src/foam/nanos/auth/ProfilePictureView.js]: FOAM style is `if ( this.uploadHidden ) {`

File: src/foam/nanos/auth/ProfilePictureView.js [hunk 4]
```diff
@@ -132,14 +192,62 @@ foam.CLASS({
...
+        return;
+      }
```
Comment (adamvy) [inline src/foam/nanos/auth/ProfilePictureView.js]: FOAM style is `} else {`

File: src/foam/nanos/auth/ProfilePictureView.js [hunk 5]
```diff
@@ -132,14 +192,62 @@ foam.CLASS({
...
+            }
+            else{
+              this.add(this.NotificationMessage.create({ message: this.FileTypeError, type: 'error' }));
+            } 
+          }
+        }else if(e.dataTransfer.files){
+          var file = e.dataTransfer.files[0];
+          if(this.isRightType(file)) 
+            this.addFile(file);
+          else{
```
Comment (adamvy) [inline src/foam/nanos/auth/ProfilePictureView.js]: No need for {} for the else case if its also only one line

File: src/foam/nanos/auth/ProfilePictureView.js [hunk 6]
```diff
@@ -132,14 +192,62 @@ foam.CLASS({
...
+            this.add(this.NotificationMessage.create({ message: this.FileTypeError, type: 'error' }));
+          }  
+        }
+      }
+    },
+    function isRightType(file){
+      if(file.type === "image/jpg" 
+      || file.type === "image/jpeg"
```
Comment (adamvy) [inline src/foam/nanos/auth/ProfilePictureView.js]: Indentation looks wrong, || should not be lined up with the if, look elsewhere in codebase for examples.

File: src/foam/nanos/auth/ProfilePictureView.js [hunk 7]
```diff
@@ -132,14 +192,62 @@ foam.CLASS({
...
+          var file = e.dataTransfer.files[0];
+          if(this.isRightType(file)) 
+            this.addFile(file);
+          else{
+            this.add(this.NotificationMessage.create({ message: this.FileTypeError, type: 'error' }));
+          }  
+        }
+      }
+    },
+    function isRightType(file){
```
Comment (adamvy) [inline src/foam/nanos/auth/ProfilePictureView.js]: Perhaps isImageType would be a better name.

Better yet do we really care what type of file it is?  Maybe just checking that it starts with image/ is sufficient.

Comment (waseem-chamaa) [reply src/foam/nanos/auth/ProfilePictureView.js]: there are restrictions such as we are not accepting svg or gif

File: src/foam/nanos/auth/ProfilePictureView.js [hunk 8]
```diff
@@ -59,71 +66,121 @@ foam.CLASS({
...
-                    if ( sessionId ) {
+                    if ( sessionId ) 
                       url += '?sessionId=' + sessionId;
-                    }
                     return url;
                   }
-                } else {
-                   return 'images/person.svg'
-                }
+                } else return 'images/person.svg'
```
Comment (adamvy) [inline src/foam/nanos/auth/ProfilePictureView.js:151]: return should be on the next line

File: src/foam/nanos/auth/ProfilePictureView.js [hunk 9]
```diff
@@ -133,13 +190,57 @@ foam.CLASS({
...
+          var file = e.dataTransfer.files[0];
+          if( this.isImageType(file) ) this.addFile(file);
+          else 
+            this.add(this.NotificationMessage.create({ message: this.FileTypeError, type: 'error' })); 
+        }
+      }
+    },
+
+    function isImageType(file) {
+      if( file.type === "image/jpg"  ||
```
Comment (adamvy) [inline src/foam/nanos/auth/ProfilePictureView.js:227]: You should put a space after the if.

File: src/foam/nanos/auth/ProfilePictureView.js [hunk 10]
```diff
@@ -133,13 +190,57 @@ foam.CLASS({
...
+          else 
+            this.add(this.NotificationMessage.create({ message: this.FileTypeError, type: 'error' })); 
+        }
+      }
+    },
+
+    function isImageType(file) {
+      if( file.type === "image/jpg"  || 
+          file.type === "image/jpeg" || 
+          file.type === "image/png" ) return true;
```
Comment (adamvy) [inline src/foam/nanos/auth/ProfilePictureView.js:229]: Put this return on the next line, it's easy to miss otherwise

**Comment by adamvy** [comment]
A few retroactive style comments.
